packages/core/src/actors.ts

```diff
@@ -184,31 +195,34 @@ export function fromPromise<T>(
               self.send({ type: rejectEventType, data: errorData });
             }
           );
-          return undefined;
+          return state;
         case resolveEventType:
           self._parent?.send(
-            toSCXMLEvent(doneInvoke(name, event.data) as any, {
+            toSCXMLEvent(doneInvoke(name, _event.data.data) as any, {
               origin: self
             })
           );
-          return event.data;
+          return { ...state, data: event.data };
         case rejectEventType:
-          const errorEvent = error(name, event.data);
+          const errorEvent = error(name, _event.data.data);
 
           self._parent?.send(
             toSCXMLEvent(errorEvent, {
               origin: self
             })
           );
-          return event.data;
+          return { ...state, data: event.data };
         case stopSignalType:
-          canceled = true;
-          return undefined;
+          return { ...state, canceled: true };
         default:
-          return undefined;
+          return state;
       }
     },
-    initialState: undefined
+    initialState: {
+      canceled: false,
+      data: undefined
+    },
+    getSnapshot: (state) => state.data
```

Comment:
This ensures that `.getSnapshot()` is either the resolved data or `undefined` instead of _all_ of the internal state